DigitalLearningSolutions.Data.Tests/Services/SessionServiceTests.cs

```diff
@@ -48,13 +48,27 @@ public void Index_should_StartOrRestartSession_for_course_not_in_session()
                 .MustNotHaveHappened();
 
             A.CallTo(() => sessionDataService.UpdateSessionDuration(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void StartOrUpdateSession_should_add_session_to_context_for_course_not_in_session()
+        {
+            // Given
+            httpContextSession.Clear();
+            const int newCourseId = CustomisationId;  // Not in session
+            const int oldCourseInSession = CustomisationId + 1;
+            httpContextSession.SetInt32($"SessionID-{oldCourseInSession}", DefaultSessionId + 1);
+
+            // When
+            sessionService.StartOrUpdateSession(CandidateId, newCourseId, httpContextSession);
 
+            // Then
             httpContextSession.Keys.Should().BeEquivalentTo($"SessionID-{newCourseId}");
             httpContextSession.GetInt32($"SessionID-{newCourseId}").Should().Be(DefaultSessionId);
         }
 
         [Test]
-        public void Index_should_UpdateSession_for_course_in_session()
+        public void StartOrUpdateSession_should_UpdateSession_for_course_in_session()

```

Comment:
This test also needs to be split up 🙂

Reply:
This was actually split in two, I just forgot to delete the second part of the assertions! Fixed in 1cd9ace 🙂